packages/@react-aria/menu/src/useSafelyMouseToSubmenu.ts

```diff
@@ -0,0 +1,90 @@
+import {CSSProperties, RefObject, useEffect, useRef, useState} from 'react';
+import {useResizeObserver} from '@react-aria/utils';
+
+interface SafelyMouseToSubmenuOptions {
+  /** Ref for the submenu. */
+  submenuRef: RefObject<Element>,
+  /** Whether the submenu is open. */
+  isOpen: boolean
+}
+
+/**
+ * Allows the user to move their pointer to the submenu without it closing when their mouse leaves the trigger element.
+ * Prevents pointer events from going to the underlying menu if the user is moving their pointer towards the sub-menu.
+ */
+export function useSafelyMouseToSubmenu(options: SafelyMouseToSubmenuOptions): CSSProperties {
+  let {submenuRef, isOpen} = options;
+  let prevPointerPos = useRef<{x: number, y: number}>(null);
+  let submenuRect = useRef<DOMRect>(null);
+  let lastProcessedTime = useRef<number>(0);
+  let [isPointerMovingTowardsSubmenu, setIsPointerMovingTowardsSubmenu] = useState(false);
+
+  let updateSubmenuRect = () => {
+    if (submenuRef.current) {
+      submenuRect.current = submenuRef.current.getBoundingClientRect();
+    }
+  };
+
+  useResizeObserver({ref: submenuRef, onResize: updateSubmenuRect});
+
+  useEffect(() => {
+    let submenu = submenuRef.current;
+
+    if (!submenu || !isOpen) {
+      setIsPointerMovingTowardsSubmenu(false);
+      return;
+    }
+
+    submenuRect.current = submenu.getBoundingClientRect();
+
+    let onPointerMove = (e: PointerEvent) => {
```

Comment:
just a heads up, but this will fire on a touch device if the user is scrolling a menu causing the following bug: 
"on iPad, if you open a submenu and then scroll it, you cannot tap on another item in the parent menu to close the submenu anymore". Digging for a good way to only apply this for mouse movement...

Reply:
The most straight forward way seems to be to just check `e.pointerType` and if it is touch then early return. Verified that `e.pointerType=touch` on my Android device and test iPad

Reply:
Will do, good find!